Library/Homebrew/test/utils/analytics_spec.rb

```diff
@@ -0,0 +1,99 @@
+require "utils/analytics"
+require "formula_installer"
+
+describe Utils::Analytics do
+  describe "::os_prefix_ci" do
+    context "when anonymous_os_prefix_ci is not set" do
+      before(:each) do
+        described_class.clear_anonymous_os_prefix_ci_cache
+      end
+
+      it "returns OS_VERSION and prefix when HOMEBREW_PREFIX is not /usr/local" do
+        stub_const("HOMEBREW_PREFIX", "blah")
+        expect(described_class.os_prefix_ci).to include("#{OS_VERSION}, non-/usr/local")
+      end
+
+      it "includes CI when ENV['CI'] is set" do
+        ENV["CI"] = "true"
+        expect(described_class.os_prefix_ci).to include("CI")
+      end
+
+      it "does not include prefix when HOMEBREW_PREFIX is usr/local" do
+        stub_const("HOMEBREW_PREFIX", "/usr/local")
+        expect(described_class.os_prefix_ci).not_to include("non-/usr/local")
+      end
+    end
+  end
+
+  describe "::report_event" do
+    let(:f) { formula { url "foo-1.0" } }
+    let(:options) { FormulaInstaller.new(f).display_options(f) }
+    let(:action)  { "#{f.full_name} #{options}".strip }
+
+    context "when ENV vars is set" do
+      it "returns nil when HOMEBREW_NO_ANALYTICS is true" do
+        ENV["HOMEBREW_NO_ANALYTICS"] = "true"
+        expect(described_class.report_event("install", action)).to be_nil
+      end
+
+      it "returns nil when HOMEBREW_NO_ANALYTICS_THIS_RUN is true" do
+        ENV["HOMEBREW_NO_ANALYTICS_THIS_RUN"] = "true"
+        expect(described_class.report_event("install", action)).to be_nil
+      end
+
+      it "returns nil when HOMEBREW_ANALYTICS_DEBUG is true" do
+        ENV.delete("HOMEBREW_NO_ANALYTICS_THIS_RUN")
+        ENV.delete("HOMEBREW_NO_ANALYTICS")
+        ENV["HOMEBREW_ANALYTICS_DEBUG"] = "true"
+        expect(described_class.report_event("install", action)).to be_nil
+      end
+    end
+
+    context "when ENV vars are nil" do
+      before do
+        ENV.delete("HOMEBREW_NO_ANALYTICS_THIS_RUN")
+        ENV.delete("HOMEBREW_NO_ANALYTICS")
+      end
+
+      it "returns waiting thread when HOMEBREW_ANALYTICS_DEBUG is not set" do
```

Comment:
I don't think anything actually uses this - I believe this is just an implementation detail. We don't need to test that.